inmoose/pycombat/covariates.py

```diff
@@ -150,6 +158,98 @@ def __init__(self, message):
         super().__init__(self.message)
 
 
+def format_covar_mod(covar_mod, batch, cov_missing_value=None):
+    """Format the covariate table in model matrix.
+
+    Arguments
+    ---------
+    covar_mod : list or matrix
+        model matrix (dataframe, list or numpy array) contaning one or multiple covariates
+    batch : array or list or :obj:`inmoose.utils.factor.Factor`
+        batch indices
+    cov_missing_value : str
+        Option to choose the way to handle missing covariates
+        `None` raise an error if missing covariates and stop the code
+        "remove" remove samples with missing covariates and raise a warning
+        "fill" handle missing covariates, by creating a distinct covariate per batch
+        (default: "raise")
+
+    Returns
+    -------
+    remove_sample : list
+        sample indices to remove from the analysis
+    covar_mod : dataframe
+        model matrix (dataframe) contaning covariates in interger format
+    """
+    if (type(covar_mod) == list) | (type(covar_mod) == np.ndarray):
```

Comment:
What about checking that covar_mod is a pandas dataframe and return an error if it's not one of the 3 types?

Reply:
Good idea, it's done !